### DISCUSSION TITLE
Intermittent timeouts on GitHub Actions

### DISCUSSION THREAD

**Discussion by diaz-hfc** [cypress-io/cypress#27423] [Questions and Help]
I am using cypress-io/github-action@v5 to run my Cypress tests on GitHub actions, using headless Chrome, but for some reason from 20 runs, 2 fails due to timeout errors, see stack below:

```
"before all" hook for "should have links with source_code from the URL params":
     CypressError: Timed out after waiting `60000ms` for your remote page to load.

Your page did not fire its `load` event within `60000ms`.

You can try increasing the `pageLoadTimeout` value in `cypress.config.ts` to wait longer.

Browsers will not fire the `load` event until all stylesheets and scripts are done downloading.

When this `load` event occurs, Cypress will continue running commands.

Because this error occurred during a `before all` hook we are skipping the remaining tests in the current suite: 
```

We are running these tests against a production environment, I am not able to reproduce this on my local computer it only happens in GitHub actions.

Having some theories.

1. Connectivity issues between GH runner and our webpages (slow download maybe).
2. Issues with tests written.
3. As this is something that does not fail on my local computer, I am dropping the idea this is a Cypress issue.

I wonder if there is a way to debug with more details what is happening during the GH action to go more deeper into the problem?

**Comment by portyu9** [comment]
Hi @diaz-hfc 

The error tells us something more specific than a normal Cypress command timeout:

> the page did not fire its `load` event within 60000ms

That means I would investigate the **page/resource loading**, not the selector/test command that follows it. The browser does not fire `load` until the relevant page resources have finished loading.

Since this only happens occasionally in GitHub Actions against production, capture the failing run's:

- browser network failures / requests that remain pending;
- browser console errors;
- Cypress screenshot/video or Test Replay;
- production/CDN/WAF logs at the same timestamp.

That should show whether a script, stylesheet, redirect, CDN resource, third-party resource, etc. is hanging specifically from the GitHub runner.

I would also update the workflow. `cypress-io/github-action@v5` is now unsupported; the current recommended major is:

```yaml
- uses: cypress-io/github-action@v7
```

and I would run it with a maintained Cypress/browser combination.

Increasing `pageLoadTimeout` can be a useful **diagnostic** if the resources are simply slow, but it isn't a root-cause fix. If one resource occasionally never completes, increasing 60 seconds to 120 seconds only postpones the same failure.

Please mark this answer as accepted if it helped, thank you.
